extensions/policy-addon-kubernetes/helm-charts/azure-policy-addon-arc-clusters/templates/azure-policy-patch-label.yaml

```diff
@@ -0,0 +1,123 @@
+# 
+# The job in this YAML file adds the 'control-plane' label to 'kube-system' and 'azure-arc' namespaces.
+# The job is executed in helm pre-install phase of helm life cycle.
+#
+# For namespaces with 'control-plane' label...
+#     1. Kubernetes admission requests with violations are not denied by Gatekeeper.
+#     2. Auditing results are still available.
+#
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: azure-policy-patch-label-account
+  namespace: kube-system
+  annotations:
+    "helm.sh/hook-weight": "20"
+    "helm.sh/hook": pre-install
+    "helm.sh/hook-delete-policy": before-hook-creation,hook-succeeded
+---
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: azure-policy-patch-label-cluster-role
+  annotations:
+    "helm.sh/hook-weight": "25"
+    "helm.sh/hook": pre-install
+    "helm.sh/hook-delete-policy": before-hook-creation,hook-succeeded
+rules:
+- apiGroups: [""]
+  resources: ["namespaces"]
+  verbs: ["get", "patch"]
+---
+kind: RoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: azure-policy-patch-label-role-binding
+  namespace: kube-system
+  annotations:
+    "helm.sh/hook-weight": "30"
+    "helm.sh/hook": pre-install
+    "helm.sh/hook-delete-policy": before-hook-creation,hook-succeeded
+subjects:
+- kind: ServiceAccount
+  name: azure-policy-patch-label-account
+  namespace: kube-system
+roleRef:
+  kind: ClusterRole
+  name: azure-policy-patch-label-cluster-role
+  apiGroup: rbac.authorization.k8s.io
+---
+kind: RoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: azure-policy-patch-label-role-binding
+  namespace: azure-arc
+  annotations:
+    "helm.sh/hook-weight": "30"
+    "helm.sh/hook": pre-install
+    "helm.sh/hook-delete-policy": before-hook-creation,hook-succeeded
+subjects:
+- kind: ServiceAccount
+  name: azure-policy-patch-label-account
+  namespace: kube-system
+roleRef:
+  kind: ClusterRole
+  name: azure-policy-patch-label-cluster-role
+  apiGroup: rbac.authorization.k8s.io
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: azure-policy-patch-label
+  namespace: kube-system
+  annotations:
+    "helm.sh/hook-weight": "35"
+    "helm.sh/hook": pre-install
+    "helm.sh/hook-delete-policy": before-hook-creation,hook-succeeded
+spec:
+  template:
+    metadata:
+      labels:
+        app: azure-policy-patch-label
+    spec:
+      serviceAccountName: azure-policy-patch-label-account
+      automountServiceAccountToken: true
+      restartPolicy: Never
+      terminationGracePeriodSeconds: 0
+      containers:
+        - name: azure-policy-patch-label
+          image: {{ .Values.alpine.image.name }}:{{ .Values.alpine.image.tag }}
+          imagePullPolicy: IfNotPresent
+          command: 
+            - "/bin/sh"
+            - "-ec"
+            - |
+              set -o pipefail
+              token=$(cat /var/run/secrets/kubernetes.io/serviceaccount/token)
+              namespaces="kube-system azure-arc"
```

Comment:
This should maybe be an environment variable? Makes it easier to see and change

Reply:
For now its fine I think as this is a short-term solution anyways.